### PR TITLE
fix: trackevent enabled is undefined

### LINKED ISSUE #12117: [Bug]: metametrics enable value is undefined
### Describe the bug

When calling trackEvent, the event is not tracked as the MetaMetrics instance `enabled` value is undefined.
The value is properly retrieved by config step from storage and is true (if user opted in) on app load, but then becomes undefined.
See for instance [track event on settings](https://github.com/MetaMask/metamask-mobile/blob/d855104784230d2e5d50feaff69e9f6133d65cd7/app/components/Views/Settings/index.tsx#L57-L57), adding a log on [MetaMetrics.trackEvent](https://github.com/MetaMask/metamask-mobile/blob/d855104784230d2e5d50feaff69e9f6133d65cd7/app/core/Analytics/MetaMetrics.ts#L669-L669) shows `this.enabled` is undefined when it should be either true or false.  

### Expected behavior

`this.enabled` should always be true or false (default) but not undefined.

### Screenshots/Recordings

_No response_

### Steps to reproduce

go in settings screen, notice no event sent.

### Error messages or log output

I added `console.log('MetaMetrics.ts trackEvent() _event this.enabled:', _event, this.enabled);` on `trackEvent` and this outputs:
```
 LOG  MetaMetrics.ts trackEvent() _event this.enabled: {"category": "Navigation Drawer", "properties": {"action": "Navigation Drawer", "name": "Settings"}} undefined
```

### Detection stage

In production (default)

### Version

latest on main 1cf60a3f73b1fb08eef08dd304c73521d7fe4e12

### Build type

None

### Device

at least simulator

### Operating system

iOS

### Additional context

To be bissected and refined.
Unit tests on MetaMetrics do not detect the error.

### Severity

It's not visible for user but critical for our ability to get metrics on the app.

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

> [!IMPORTANT]
> The issue is due to the fact that `MetaMetrics.trackEvent` method has multiple signatures to handle the backward compatibility with old ways to call it.
> After discussion (see the change history of this description), we decided to remove backward compatibility and the multiple signature system. We are going to simplify and then fix all the `trackEvent` calls.
Asside from the changes in `MetaMetrics` and `useMetrics` hook (and tests) all the changes should be on the calls of `trackEvent`.

### Current changes in this PR
- removes multiple signature `MetaMetrics.trackEvent`
- updates `MetaMetrics` unit tests
- updates `useMetrics` hook
- updates `useMetrics` hook unit tests
- deletes now useless legacy compatibility utils
- updates all `trackEvent` calls
- updates all unit tests that test `trackEvent` calls

## **Related issues**

Fixes #12117 

## **Manual testing steps**

1. navigate the app
2. check trackEvent is called (check app logs)

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/058f6607-eda1-4eb5-bcae-a774deb13648

https://github.com/user-attachments/assets/fa0816cf-5459-4825-a162-a9a2e87c86ac

### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by NicolasMassart** [comment]
Converted to draft as I have issues fixing app/components/Views/Settings/SecuritySettings/Sections/MetaMetricsAndDataCollectionSection/MetaMetricsAndDataCollectionSection.test.tsx

**Comment by NicolasMassart** [comment]
We decided to go with not fixing but removing the initial cause (that is the backward compatible multi signature trackEvent), and go with a cleanup and update all calls to use new method with event builder.
All changes in this PR will be reverted after this comment.

**Comment by kylanhurt** [comment]
@NicolasMassart Unit tests failing and linting errors. You able to handle those when you get a chance?

File: app/components/UI/Notification/List/index.tsx [hunk 1]
```diff
@@ -83,23 +83,26 @@ export function NotificationsListItem(props: NotificationsListItemProps) {
...
   );
 
   const menuItemState = useMemo(() => {
     const notificationState =
-      props.notification?.type && hasNotificationComponents(props.notification.type)
-      ? NotificationComponentState[props.notification.type]
-      : undefined;
+      props.notification?.type &&
+      hasNotificationComponents(props.notification.type)
+        ? NotificationComponentState[props.notification.type]
```
Comment (kylanhurt) [inline app/components/UI/Notification/List/index.tsx:104]: Could you not use the following syntax here?

```
hasNotificationComponents(props.notification.type)
        && NotificationComponentState[props.notification.type]```

Comment (NicolasMassart) [reply app/components/UI/Notification/List/index.tsx:104]: I tried not to change things outside of the scope of just updating the trackEvent call

File: .yarnrc [hunk 2]
```diff
@@ -1 +1,6 @@
-ignore-scripts true
\ No newline at end of file
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (sethkfman) [inline .yarnrc:1]: This change should be reverted

File: .yarn/releases/yarn-1.22.22.cjs
Comment (sethkfman) [inline .yarn/releases/yarn-1.22.22.cjs:1]: Revert this change

**Comment by NicolasMassart** [comment]
So far I checked:
- all the settings screens events ✅ 
- some random events on browser ✅ 
- some random events on wallet ✅ 
- some random events on ramp process ✅

**Review by brianacnguyen — DISMISSED** [review]

**Review by wachunei — DISMISSED** [review]
Ramp changes LGTM

**Review by jclancy93 — DISMISSED** [review]

**Review by brianacnguyen — DISMISSED** [review]

**Comment by Matt561** [comment]
@NicolasMassart would it be possible to update the recently added `withMetaMetrics` hook to make sure these changes don't break our staking events before release? I suspect the lift is small since it's a single function.

I don't mind adding a fix as part of this PR too if you'd rather I do it! Just let me know.

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Comment by NicolasMassart** [comment]
> @NicolasMassart would it be possible to update the recently added `withMetaMetrics` high order function to make sure these changes don't break our staking events before release? I suspect the lift is small since it's a single function.
> 
> I don't mind adding a fix as part of this PR too if you'd rather I do it! Just let me know.

I do not see any reason why this would not work. Your wrapper was already using the new way to call trackEvent.
However, your confidence could be highly improved by simply having a unit test for this `withMetaMetrics`. So please, write one if you want to make sure your code is not impacted by this or future work.

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by wachunei — APPROVED** [review]

**Review by brianacnguyen — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by darkwing — APPROVED** [review]

**Pull request merged**